src/main/java/bio/terra/service/filedata/flight/ingest/IngestFileDirectoryStep.java

```diff
@@ -74,23 +72,18 @@ public StepResult doStep(FlightContext context) throws InterruptedException {
                     .datasetId(datasetId)
                     .loadTag(loadModel.getLoadTag());
                 fileDao.createDirectoryEntry(dataset, newEntry);
-            } else {
-                if (!StringUtils.equals(existingEntry.getLoadTag(), loadModel.getLoadTag())) {
-                    // (a) Exists and is not our file
-                    throw new FileAlreadyExistsException("Path already exists: " + targetPath);
-                }
-
-                // (b) or (c) Load tags match - check file ids
+            } else if (StringUtils.equals(existingEntry.getLoadTag(), loadModel.getLoadTag())) {
```

Comment:
I think I see what you mean! So if another user creates the file entry in between the two steps, it would cause the second step to incorrectly succeed. In that case, would it make sense to keep it as one step and update the `undo`?

Reply:
Just to back up, the original problem (I think) was this:

- `doStep()` either creates a new entry (if `existingEntry` is `null`) or it reuses an existing entry
- `undoStep()` always deletes. We want it to only delete if it created the new entry. Otherwise we'll be undoing the work of a step in a flight that already completed successfully.

So the fix (in theory) was to check for the reuse case as a separate step, and store the result of the check in `FlightContext`. So when this step's undo is called, it can use the flight context state to determine if it should delete or not.

The issue I'm raising here is a different point I think. But now that I think about this more, I'm not sure if this is an issue or not. The DR code doesn't allow simultaneous loads on the same load tag. So there can't be another flight running at the same time that's also ingesting to the same load tag. I'm not 100% sure if that prevents state issues like I mentioned but it might.